File: HelpDesk-BE/HelpdeskSystem.Domain/Entities/AuditLog.cs
using HelpdeskSystem.Common.Enums;
using HelpdeskSystem.Domain.Common;

namespace HelpdeskSystem.Domain.Entities;

public class AuditLog : BaseEntity
{
    public int Id { get; set; }
    public int TicketId { get; set; }
    public int? ChangedBy { get; set; }
    public AuditAction Action { get; set; }
    public string? FieldName { get; set; }
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public string? Reason { get; set; }

    // Navigation properties
    public Ticket Ticket { get; set; } = null!;
    public User? ChangedByUser { get; set; } = null!;
}

File: HelpDesk-FE-React/src/interfaces/audit-log.ts
export interface AuditLog {
    id: number;
    ticketId: number;
    changedBy: number;
    changedByName: string;
    action: string;
    fieldName: string | null;
    oldValue: string | null;
    newValue: string | null;
    reason: string | null;
    createdAt: string;
}